### PR TITLE
Controller Changes + Profile Completion

### PR DESCRIPTION
**Pull request by tlacasse**

### CODE REVIEW

File: src/main/java/medicaldoctor/controllers/NewUserController.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        mask = 0b11;
+    }
+    
+    private void makeNewUser() throws Exception{
+        new_user_request.firstName = textFirstName.getText();
+        new_user_request.lastName = textLastName.getText();
+        new_user_request.userType = User.byName(((String)employeeTypeChoiceBox.getValue()).toUpperCase());
+        new_user_request.officeNum = Integer.parseInt(isInteger(textOfficeNumber.getText()));
+        new_user_request.department = textDepartment.getText();
+        UserService.createNewUser(new_user_request);
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/NewUserController.java]: need to display results: username and temporary password and such.

File: src/main/java/medicaldoctor/controllers/NewUserController.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    
+    HashMap<TextField, Label> textfeild_hash = new HashMap<>();
+    
+    NewUserRequest new_user_request = new NewUserRequest();
+    
+    int mask = 0b11;
+
+    @FXML
+    void submitButtonClick(ActionEvent event) throws Exception{
+        if(isFieldsEmpty(textfeild_hash)){ mask &= 0b10;}
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/NewUserController.java]: in no way should this work require bitmasks

File: src/main/java/medicaldoctor/controllers/PatientRegistrationController.java [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+            Patient patient = new Patient();
+            PatientVisit visit = new PatientVisit();
+            PatientPrescription prescription = new PatientPrescription();
+            makePatient(patient, visit, prescription);
+        }
+        mask = 0b11111;
+    }
+
+    void makePatient(Patient patient, PatientVisit visit, PatientPrescription prescription) {
+        patient.setFirstName(textFirstName.getText());
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/controllers/PatientRegistrationController.java]: i have these services

File: src/main/java/medicaldoctor/entities/UserType.java [hunk 4]
```diff
@@ -113,5 +115,12 @@ public boolean equals(Object obj) {
     public String getDashboardName(){
         return this.dashboardName;
     }
-
+    
+    public static User byUsername(String username) {
```
Comment (tlacasse) [inline src/main/java/medicaldoctor/entities/UserType.java]: remove

Comment (tlacasse) [reply src/main/java/medicaldoctor/entities/UserType.java]: this is already on the User class

File: src/main/java/medicaldoctor/controllers/NewUserController.java (hunk 2 above)
Comment (NotSalH) [reply src/main/java/medicaldoctor/controllers/NewUserController.java]: It kept short-circuiting I  forgot I can use bitwise & instead

**Pull request merged**